server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -226,6 +219,112 @@ void doRun() {
         }
     }
 
+    private List<TaskCancellation> addElapsedTimeBasedTaskCancellations(
+        List<TaskCancellation> taskCancellations,
+        List<CancellableTask> searchTasks,
+        List<CancellableTask> searchShardTasks
+    ) {
+        final Optional<TaskResourceUsageTrackers.TaskResourceUsageTracker> searchTaskElapsedTimeTracker =
+            getTaskResourceUsageTrackersByType(SearchTask.class).getElapsedTimeTracker();
+        final Optional<TaskResourceUsageTrackers.TaskResourceUsageTracker> searchShardTaskElapsedTimeTracker =
+            getTaskResourceUsageTrackersByType(SearchShardTask.class).getElapsedTimeTracker();
```

Comment:
This logic can again be agnostic of `SearchTask` and `SearchShardTask`. Same for cpu and heap based resources

Reply:
Let me think try how can we do that